### PR TITLE
More pure4: favorites/git

### PR DESCRIPTION
**Pull request by chrisnojima**
More pure actions in favorites / git
Also fixes a bug in git where if you choose 'new team' in create team repo it sends you to the teams page with the new team dialog

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers

File: shared/actions/favorite.js [hunk 1]
```diff
@@ -164,52 +164,40 @@ function _getFavoritesRPCToFolders(
   return folders
 }
 
-function* _addSaga(action: FavoriteGen.FavoriteAddPayload): Saga.SagaGenerator<any, any> {
+function* _addOrIgnoreSaga(
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: Why can't this be a pure fn? You can either use your error handler or CallAndWrap

File: shared/actions/favorite.js [hunk 2]
```diff
@@ -164,52 +164,40 @@ function _getFavoritesRPCToFolders(
...
-      yield put(FavoriteGen.createFavoriteList())
-    } catch (error) {
-      console.warn('Err in favorite.favoriteIgnore', error)
+      console.warn('Err in favorite.favoriteAddOrIgnore', error)
     }
   }
 }
 
 function* _listSaga(): Saga.SagaGenerator<any, any> {
+  const state: TypedState = yield select()
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: This can also be pure, no?

File: shared/actions/git.js [hunk 3]
```diff
@@ -155,27 +149,26 @@ const _onTabChange = (action: RouteTreeTypes.SwitchTo) => {
...
-  yield Saga.safeTakeLatest(GitGen.setError, _setError)
+  yield Saga.safeTakeEveryPure(GitGen.createPersonalRepo, _createPersonalRepo)
+  yield Saga.safeTakeEveryPure(GitGen.createTeamRepo, _createTeamRepo)
+  yield Saga.safeTakeEveryPure(GitGen.deletePersonalRepo, _deletePersonalRepo)
+  yield Saga.safeTakeEveryPure(GitGen.deleteTeamRepo, _deleteTeamRepo)
+  yield Saga.safeTakeLatestPure(GitGen.setLoading, _setLoading)
+  yield Saga.safeTakeLatestPure(GitGen.setError, _setError)
   yield Saga.safeTakeEveryPure(GitGen.badgeAppForGit, _badgeAppForGit)
-  yield Saga.safeTakeEvery(GitGen.handleIncomingGregor, _handleIncomingGregor)
+  yield Saga.safeTakeEveryPure(GitGen.handleIncomingGregor, _handleIncomingGregor)
```
Comment (MarcoPolo) [inline shared/actions/git.js]: 💥

File: shared/git/new-repo/container.js [hunk 4]
```diff
@@ -24,7 +24,10 @@ const mapDispatchToProps = (dispatch: any, {navigateAppend, navigateUp, routePro
       : GitGen.createCreatePersonalRepo({name})
     dispatch(createAction)
   },
-  onNewTeam: () => dispatch(navigateTo([teamsTab], ['showNewTeamDialog'])),
+  onNewTeam: () => {
+    dispatch(navigateTo(['showNewTeamDialog'], [teamsTab]))
```
Comment (MarcoPolo) [inline shared/git/new-repo/container.js]: Couldn't you just do `navigateTo([teamsTab, 'showNewTeamDialog'])`

Comment (chrisnojima) [reply shared/git/new-repo/container.js]: fixed, tnx!

File: shared/actions/favorite.js (hunk 2 above)
Comment (chrisnojima) [reply shared/actions/favorite.js]: yeah, but skipping for now if thats ok

**Comment by chrisnojima** [comment]
@MarcoPolo is it ok for me to merge this

**Pull request merged**